numpy/core/src/multiarray/buffer.c

```diff
@@ -276,15 +289,20 @@ _buffer_format_string(PyArray_Descr *descr, _tmp_string_t *str,
             /* Insert padding manually */
             if (*offset > new_offset) {
                 PyErr_SetString(
-                    PyExc_ValueError,
-                    "dtypes with overlapping or out-of-order fields are not "
-                    "representable as buffers. Consider reordering the fields."
-                );
+                    PyExc_ValueError, "The buffer interface does not support "
+                                      "overlapping fields or out-of-order "
+                                      "fields");
```

Comment:
Is this change of message deliberate, or just a consequence of the merge?

Reply:
deliberate, based on your review above.

I just did a little bit more than a simple rebase. I also added support for 0-sized unnamed padding, see the release note.

Reply:
oh I see what you mean, the previous message seems good enough.